src/client/service-worker/events/caching.js

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2020 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import {
+  expiration,
+  cacheRouter,
+  createMiddlewareContext,
+  getMetaData,
+  setMetaData,
+  remove,
+} from '@americanexpress/one-service-worker';
+
+// The RegExp codex:
+// - for each RegExp, we are expecting a valid http url and each of the three are "anchored"
+// -> by a pattern that starts with either "modules" (for modules and lang packs) or "app"
+// - (?<capture-group-name>.*) for an exactly labeled object made from matching parts
+// -> /(capture groups)/ of a string url. we can use `url.match(regExp).groups`to get
+// -> all the named capture groups
+// - /^https?(?::\/\/)/ start pattern anchors the start of the url and is not captured
+// -> ( /(?:.*)/ ), only matched
+// - specifying what we want to capture in (?<version>[^/]+)
+// -> we use /[^/]+/ to accurately capture what in between the slashes through negation ([^]).
+// -> for each named capture, we want to get specific on what we expect that match to be.
+// - we have a match-all in the center of the string (/https?(?::\/\/)HERE .* HERE\/(?<name>)/)
+// -> because it allows us to ignore any matches before the url is
+// -> matched from the end of the string and back
+// - if you run across `/(?<name> .{1,5})\1/`, the `\1` matches what was first matched
+// -> in a capture group. for this example: `name = \1`
+
+// groups: (modules/sha?) name / version / resource . bundle ? revision
+const moduleRegExp = /^https?(?::\/\/).*\/modules\/(?:[a-z0-9]+\/)?(?<name>[^/]+)\/(?<version>[^/]+)\/(?<resource>([a-zA-Z-~]+|\1)(?:\.\1)?)(?:\.chunk)?\.(?<bundle>(legacy\.)?browser)\.js(?:\?clientCacheRevision=(?<revision>[^/&]*))?$/;
```

Comment:
I don't think we can rely on modules being hosted under a module directory. This would have to be documented as a requirement for running one-app and modules in production which i think defeats the strength of having urls within the module map.

The same comment goes for the other RegExps